packages/network/src/api/utils.ts

```diff
@@ -0,0 +1,30 @@
+import { THUMBNAIL_RESIZE_RATIO, Image } from '@monkvision/types';
+
+function getResizedDimension(originalDimension: number): number {
+  return Math.round(originalDimension * THUMBNAIL_RESIZE_RATIO);
+}
+
+/**
+ * Params of getThumbnailUrl function.
+ */
+export type GetThumbnailUrlParams = {
+  thumbnailDomain: string;
+} & ({ image: Image } | Pick<Image, 'path' | 'width' | 'height'>);
```

Comment:
Usually, when you have a highly modular type like that for the parameters of your function, it means that your function is adapting to its usage across the project. Sometimes it is intended. But for small standalone util functions like that, I think it should be the other way around :

- You have a small standalone util function that does one specific job (here it is returning the thumbnail URL for a given image)
- Simply think about what this function needs to return the expected result. Here, it just needs the thumbnail domain, the path, width and height of the image and that's it
- This gives you the signature of the function :

```typescript
export function getThumbnailUrl(
  thumbnailDomain: string,
  path: string,
  dimensions: PixelDimensions,
): string {
  ...
}
```

Then, in your project, the different components that use your util function will have to adapt to the given parameters. Of course this is not a perfect rule that we have to apply every time, but for small standalone functions like these I think it's easier to understand.